crates/bevy_ecs/src/query/filter.rs

```diff
@@ -739,3 +739,18 @@ impl_tick_filter!(
     ChangedFetch,
     ComponentTicks::is_changed
 );
+
+pub trait ArchetypeFilter {}
```

Comment:
It's not trivial to understand what this trait does in a wider context. I would add documentation as to what for this is used. (pointing to the `ExactSizeIterator` implementation for `QueryIter`)